internal/checker/relater.go

```diff
@@ -365,10 +365,12 @@ func (c *Checker) checkTypeRelatedToEx(
 	headMessage *diagnostics.Message,
 	diagnosticOutput *[]*ast.Diagnostic,
 ) bool {
-	relaterCount := len(c.relaters)
-	c.relaters = slices.Grow(c.relaters, 1)[:relaterCount+1]
-	r := &c.relaters[relaterCount]
-	r.c = c
+	r := c.freeRelater
+	if r == nil {
+		r = &Relater{c: c}
```

Comment:
Honestly I feel like we should probably clean this up into a plain sync pool, but, not in this PR.